cmake/modules/extensions.cmake

```diff
@@ -2021,16 +2027,21 @@ function(check_set_compiler_property)
   list(REMOVE_AT COMPILER_PROPERTY_PROPERTY 0)
 
   foreach(option ${COMPILER_PROPERTY_PROPERTY})
+    if(${option} MATCHES "^SHELL:")
+      string(REGEX REPLACE "^SHELL:" "" option ${option})
+      separate_arguments(option UNIX_COMMAND ${option})
+    endif()
```

Comment:
This feels like a hack to me, but I guess this is the best we can do with the limitations of the CMake language ...

Reply:
Depends on how you look at it.
It's the same principle CMake itselfs uses, so if it is good enough to them, I assume we can use the same principle:

Ref: https://cmake.org/cmake/help/latest/command/add_compile_options.html#option-de-duplication
> One may specify a group of options using shell-like quoting along with a SHELL: prefix. The SHELL: prefix is dropped, and the rest of the option string is parsed using the [separate_arguments()](https://cmake.org/cmake/help/latest/command/separate_arguments.html#command:separate_arguments) UNIX_COMMAND mode.